Automattic/wp-calypso
# Social Previews | Update Mastodon and Tumblr previews

Fixes https://github.com/Automattic/jetpack-reach/issues/228

## Proposed Changes

* Updates Mastadon preview to the latest version
* Fixes Post preview for Mastodon preview when no media is supplied
* Fixes Post preview for Tumblr preview when no media is supplied

## Testing Instructions



* For a Jetpack site with Jetpack Advanced plan, create a post.
* Open Jetpack or Social sidebar and add custom media (photo/video)
* Enable "Share as a social post"
* Publish the post
* Boot up this PR via the quick links below
* Select the site and goto posts
* For the above published post, click on options (3 dots) and select "Share"
* Click on Preview
* Goto Mastodon section
* Confirm that the "Your post" is displayed as a media post and not as link preview
* Now uncheck "Share as a social post" for the above post and update
* Open Social Previews again as above
* Goto Mastodon section
* Confirm that the "Your post" is displayed as link preview
* Test these changes in Jetpack via https://github.com/Automattic/jetpack/pull/36874


## Pre-merge Checklist




| Header | Header |
|--------|--------|
| <img width="696" alt="Screenshot 2024-04-12 at 5 10 15 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/fb3156da-ff75-4800-9720-3879d3edd483"> | <img width="694" alt="Screenshot 2024-04-12 at 5 10 24 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/0624f0a2-ce73-47c6-9e8a-54bda7490498"> |
| <img width="696" alt="Screenshot 2024-04-12 at 5 10 43 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/b8171f50-1d65-43ba-99f8-5d1863e4d182"> | <img width="692" alt="Screenshot 2024-04-12 at 5 11 01 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/7f0477b6-2082-4b4e-96ff-8888fc475091"> |
| <img width="643" alt="Screenshot 2024-04-12 at 5 11 19 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/d60ac64d-8937-4c23-86d3-b070ac411fd7"> | <img width="641" alt="Screenshot 2024-04-12 at 5 12 03 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/e2c3a75c-19f9-4fab-b918-833f9cc98d2e"> |
